System/GraphicalPrimitive.cpp

```diff
@@ -371,33 +390,114 @@ namespace RTE {
 		}
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PolygonPrimitive::Draw(BITMAP *drawScreen, const Vector &targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart;
+			Vector drawEnd;
+			for (int i = 0; i < m_Vertices.size(); ++i) {
+				drawStart = m_StartPos - targetPos + (*m_Vertices[i]);
+				drawEnd = m_StartPos - targetPos + ((i + 1 < m_Vertices.size()) ? *m_Vertices[i + 1] : *m_Vertices[0]);
+				line(drawScreen, drawStart.GetFloorIntX(), drawStart.GetFloorIntY(), drawEnd.GetFloorIntX(), drawEnd.GetFloorIntY(), m_Color);
+			}
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+			Vector drawEndLeft;
+			Vector drawEndRight;
+			for (int i = 0; i < m_Vertices.size(); ++i) {
+				TranslateCoordinates(targetPos, m_StartPos + (*m_Vertices[i]), drawStartLeft, drawStartRight);
+				TranslateCoordinates(targetPos, m_StartPos + ((i + 1 < m_Vertices.size()) ? *m_Vertices[i + 1] : *m_Vertices[0]), drawEndLeft, drawEndRight);
+
+				line(drawScreen, drawStartLeft.GetFloorIntX(), drawStartLeft.GetFloorIntY(), drawEndLeft.GetFloorIntX(), drawEndLeft.GetFloorIntY(), m_Color);
+				line(drawScreen, drawStartRight.GetFloorIntX(), drawStartRight.GetFloorIntY(), drawEndRight.GetFloorIntX(), drawEndRight.GetFloorIntY(), m_Color);
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PolygonFillPrimitive::Draw(BITMAP *drawScreen, const Vector &targetPos) {
+		size_t drawPointsSize = m_Vertices.size() * 2;
+
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+
+			std::vector<int> drawPoints = {};
+			drawPoints.reserve(drawPointsSize);
+
+			for (const Vector *vertice : m_Vertices) {
+				drawPoints.insert(drawPoints.end(), { drawStart.GetFloorIntX() + vertice->GetFloorIntX(), drawStart.GetFloorIntY() + vertice->GetFloorIntY() });
+			}
+			polygon(drawScreen, m_Vertices.size(), drawPoints.data(), m_Color);
+		} else {
+			std::vector<int> drawPointsLeft = {};
+			drawPointsLeft.reserve(drawPointsSize);
+
+			std::vector<int> drawPointsRight = {};
+			drawPointsRight.reserve(drawPointsSize);
+
+			Vector drawPointLeft;
+			Vector drawPointRight;
+			for (const Vector *vertice : m_Vertices) {
+				TranslateCoordinates(targetPos, m_StartPos + (*vertice), drawPointLeft, drawPointRight);
+
+				drawPointsLeft.insert(drawPointsLeft.end(), { drawPointLeft.GetFloorIntX(), drawPointLeft.GetFloorIntY() });
```

Comment:
Why use insert here instead of push back or emplace back?

Reply:
Because i'm inserting 2 elements at once.

Reply:
Doesn't emplace back happily accept multiple values at once? Or is that only when emplacing back to maps or whatever?